### ISSUE TITLE
No Docker client strategy found

### ISSUE THREAD

**Issue by levitomer** [testcontainers/testcontainers-node#321]
My build in travis is breaking with the error `No Docker client strategy found`

version: "testcontainers": "8.1.2"

```
  FAIL All test/app.integration.ts (10.2 s)
  Setup environment successfuly › Connects to kafka
    No Docker client strategy found
      at node_modules/testcontainers/dist/docker/docker-client.js:47:11
      at fulfilled (node_modules/testcontainers/dist/docker/docker-client.js:5:58)```

**Comment by cristianrgreco** [comment]
This issue has inspired me to create an issue template 😄 Please share logs (`DEBUG=testcontainers*`), node version, docker version, the code, etc.

**Comment by gareth-cheeseman** [comment]
I have the same error -  at least for me I think it is a combination of a few things:

- the code being run in a docker build
- changes in node 15 to error on unhandled rejections, instead of warn
- running tests with `NODE_OPTIONS=--unhandled-rejections=warn` avoids the error stoping the docker build, or running with a node 14 docker image as base.

I have jest tests using testcontainers that are skipped if the tests are run inside a docker container.
The set up code for testcontainers is in a separate module that's imported in the test file, so the code is evaluated even though the tests are skipped.

**Logs** 
testcontainers DEBUG Found applicable Docker client strategy: UnixSocketStrategy
testcontainers DEBUG Testing Docker client strategy URI: unix:///var/run/docker.sock
testcontainers WARN  Docker daemon is not reachable: Error: connect ENOENT /var/run/docker.sock
testcontainers WARN  Docker client strategy UnixSocketStrategy is not reachable

**Node version**: 16 latest
**Docker version**: 20.10.12

I'll try to provide a replicating example later.

**Comment by cristianrgreco** [comment]
There is not enough information in this issue to be able to investigate. Pls feel free to raise a new issue with more information, reproduction steps, etc, if the issue persists. @PSanetra mentioned a dependency upgrade fixed the issue, since v8.5.1, all dependencies have had their patch versions upgraded.
